src/debug/debug.cc

```diff
@@ -1229,17 +1230,17 @@ debugLogTime(const time_t t)
         // 4 bytes smaller than buf to ensure .NNN catenation by snprintf()
         // is safe and works even if strftime() fills its buffer.
         char buf2[sizeof(buf)-4];
-        tm = localtime(&t);
+        tm = localtime(&(t.tv_sec));
         strftime(buf2, sizeof(buf2), "%Y/%m/%d %H:%M:%S", tm);
         buf2[sizeof(buf2)-1] = '\0';
-        const int sz = snprintf(buf, sizeof(buf), "%s.%03d", buf2, static_cast<int>(current_time.tv_usec / 1000));
+        const auto sz = snprintf(buf, sizeof(buf), "%s.%03d", buf2, static_cast<int>(t.tv_usec / 1000));
         assert(0 < sz && sz < static_cast<int>(sizeof(buf)));
-        last_t = t;
-    } else if (t != last_t) {
-        tm = localtime(&t);
+        last_t = t.tv_sec;
```

Comment:
This line corresponds to an annoying bug in the official code -- it leaves `buf` dirty for up to a second after level-2+ output. Since we are modifying this line, I suggest fixing this bug as well:

```suggestion

        // force buf reset for subsequent level-0/1 messages that should have no milliseconds
        last_t = 0;
```

... and add "Also fix bug that was adding stale milliseconds value to some level-0/1 messages" or similar paragraph the PR description.

The above suggested fix is untested. Please test.

Reply:
Done.